### PR TITLE
Fix data loss in ekf2 and ekf2 replay and enable smaller buffers to reduce RAM

### PR DESCRIPTION
**Pull request by priseborough**
Fixes a time stamping bug in the ekf2_replay app that could result in repeated mag and baro data being used.

Allows high rate magnetometer and baro data to be used without data loss by downsampling data using a running average if the data rate is too high. This will allow a smaller observation buffer size to be used on the ekf2 to save RAM by setting the delay parameters for all sensors not required to zero and tuning the remaining delay parameters using replay logs.

This has been tested on replay and flight (copter only) in combination with https://github.com/PX4/ecl/pull/213

### CODE REVIEW

File: src/modules/ekf2/ekf2_main.cpp [hunk 1]
```diff
@@ -135,6 +135,27 @@ class Ekf2 : public control::SuperBlock
...
+	uint64_t _timestamp_mag_us = 0;
+	uint64_t _timestamp_balt_us = 0;
+
+	// Used to down sample magnetometer data
+	float _mag_data_sum[3];			// summed magnetometer readings (Ga)
+	uint64_t _mag_time_sum_ms;		// summed magnetoemter time stamps (msec)
+	uint8_t _mag_sample_count = 0;		// number of magnetometer measurements summed
+	uint32_t _mag_time_ms_last_used = 0;	// time stamp in msec of the last averaged magnetometer measurement used by the EKF
+
+	int	_sensors_sub = -1;
```
Comment (bkueng) [inline src/modules/ekf2/ekf2_main.cpp]: All of these *_sub variables are no longer needed, they're defined in `Ekf2::task_main` now.

Comment (priseborough) [inline src/modules/ekf2/ekf2_main.cpp]: Will fix

**Comment by priseborough** [comment]
Flight tested on a pixracer quad at commit 97362ac with EKF2_MIN_OBS_DT set to 50. Innovation levels were low - see below:

gps vel innov
gps pos innov
baro alt innov
mag z innov
mag y innov
mag x innov

Output predictor tracking errors were low:

output predictor tracking errors

**Comment by LorenzMeier** [comment]
@bkueng Can you remove INAV from px4fmu-v2_test config and also remove inav from the rcS startup? Anyone having set INAV should default now to LPE.

**Comment by bkueng** [comment]
@LorenzMeier I removed INAV from the param options & output a warning via mavlink if set via QGC. If currently set to INAV, the system will switch to LPE.

@priseborough is this good to be merged?

**Comment by priseborough** [comment]
@bkueng yes, once https://github.com/PX4/ecl/pull/213 is merged

**Pull request merged**